### DISCUSSION TITLE
Automatic audio resampling on the fly

### DISCUSSION THREAD

**Discussion by agemagician** [NVIDIA-NeMo/NeMo#2360] [Q&A] [answered]
Is it possible to resampling audio on the fly during training ?

Assuming I have a dataset that consists of different sampling rage "32k, 16k, and 8k", Can Nemo automatically resample them during training into the model-specific sample rate defined in the YAML file?

**Comment by okuchaiev** [comment, marked as answer]
Yes, there is a ``sample_rate`` parameter you can pass https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/api.html?highlight=AudioToCharDataset#nemo.collections.asr.data.audio_to_text.AudioToCharDataset

However, I would strongly recommend against this and instead pre-process your data into the right sampling rate before training. Otherwise it would really hurt training speed.
